### PR TITLE
Misha Seltzer	First version of Golang proxy and a Python client

### PR DESCRIPTION
**Pull request by mishas**

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -0,0 +1,46 @@
+
+new_git_repository(
+    name = "pika_git",
+    remote = "https://github.com/pika/pika.git",
+    tag = "0.10.0",
+    build_file_content = """
+py_library(
+    name = "pika",
+    srcs = glob(["pika/*.py", "pika/**/*.py"]),
```
Comment (nadirizr) [inline WORKSPACE:9]: Isn't the second glob enough to catch the main directory as well?

File: config.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+          url: "amqps://my_rabbitmq/"
+          exchange: "my_exchange"
+  
+      tls_config:
+          ca_file: "my_certs/cacert.pem"
+          cert_file: "my_certs/cert.pem"
+          key_file: "my_certs/key.pem"
+
+      target_groups:
+        - targets: ["amqps://my_rabbitmq:7149/", "amqps://my_rabbitmq:8153/"]
```
Comment (nadirizr) [inline config.yaml:14]: Maybe you should change the example numbers here... not too important, just stands out :)

File: proxy/config.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+type TargetGroup struct {
+	Targets []string
+}
+
+type ScrapeConfig struct {
+	// The job name to which the job label is set by default.
+	JobName    string     `yaml:"job_name"`
+	AMQPConfig AMQPConfig `yaml:"amqp_config,omitempty"`
+	// TLSConfig to use to connect to the targets.
+	TLSConfig TLSConfig `yaml:"tls_config,omitempty"`
```
Comment (nadirizr) [inline proxy/config.go:40]: Align this line to the ones above (f you already did that with the two lines above).

File: proxy/config.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	CertFile string `yaml:"cert_file,omitempty"`
+	// The client key file for the targets.
+	KeyFile string `yaml:"key_file,omitempty"`
+	// Used to verify the hostname for the targets.
+	ServerName string `yaml:"server_name,omitempty"`
+	// Disable target certificate validation.
+	InsecureSkipVerify bool `yaml:"insecure_skip_verify"`
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
```
Comment (nadirizr) [inline proxy/config.go:28]: Why is this important here? And if it is, why not have it everywhere? Is this standard practice for Go?

File: proxy/proxy.go [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+func (a *externalAuth) Mechanism() string {
+	return "EXTERNAL"
+}
+func (a *externalAuth) Response() string {
+	return fmt.Sprintf("\000")
+}
+
+// getAMQPConfig returns a reference to the amqp.Config object.
+// certsDir should be the path to the directory holding {cacert,cert,key}.pem files for the TLS
```
Comment (nadirizr) [inline proxy/proxy.go:28]: Fix the comment to indicate that you now use file config variables instead of a directory.

File: proxy/rpc.go [hunk 6]
```diff
@@ -0,0 +1,182 @@
+package rpc
+
+import (
+	"errors"
```
Comment (nadirizr) [inline proxy/rpc.go:4]: Is the standard tab size in Go 6 spaces??

File: config.yaml (hunk 2 above)
Comment (mishas) [inline config.yaml:14]: Nope...

File: proxy/config.go (hunk 4 above)
Comment (mishas) [inline proxy/config.go:28]: It's not go, it's an external library for parsing yaml.
This is here to make sure that our config file doesnt have unnecessary (illegal) definitions. I don't care about it myself, and this struct was copied from prometheus.

File: proxy/config.go (hunk 3 above)
Comment (mishas) [inline proxy/config.go:40]: Nope.
Go aligns stuff itself (with gofmt). Comments and blank lines break continuation.

File: proxy/proxy.go (hunk 5 above)
Comment (mishas) [inline proxy/proxy.go:28]: done

File: proxy/rpc.go (hunk 6 above)
Comment (mishas) [inline proxy/rpc.go:4]: Nope. Go uses tabs... Github shows it as 6...

**Pull request merged**